From d290448bfa838cd8f9590b841621e34c9513a86e Mon Sep 17 00:00:00 2001
From: "Nag, AbhrodeepX" <abhrodeepx.nag@intel.com>
Date: Tue, 27 Dec 2022 11:56:03 +0000
Subject: [PATCH] CTL: Updating BMC to support BIOS version 3.4

BMC used to support BIOS version 3.2 and 3.3, now post BIOS
upgrade to version 3.4 BMC also has to support BIOS version 3.4.

Tested:
Checked smbios2 file is successfully generated in the location
/var/lib/smbios and tree also generated properly.

Signed-off-by: Nag, AbhrodeepX <abhrodeepx.nag@intel.com>
Upstream-Status: Pending
---
 include/smbios_mdrv2.hpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/include/smbios_mdrv2.hpp b/include/smbios_mdrv2.hpp
index 0ac77b9..24904fb 100644
--- a/include/smbios_mdrv2.hpp
+++ b/include/smbios_mdrv2.hpp
@@ -167,8 +167,9 @@ static constexpr const char* pciePath =
 static constexpr const char* systemPath =
     "/xyz/openbmc_project/inventory/system/chassis/motherboard/bios";
 
-constexpr std::array<SMBIOSVersion, 3> supportedSMBIOSVersions{
-    SMBIOSVersion{3, 2}, SMBIOSVersion{3, 3}, SMBIOSVersion{3, 5}};
+constexpr std::array<SMBIOSVersion, 4> supportedSMBIOSVersions{
+    SMBIOSVersion{3, 2}, SMBIOSVersion{3, 3}, SMBIOSVersion{3, 4},
+	    SMBIOSVersion{3, 5}};
 
 typedef enum
 {
-- 
2.17.1

